eth-deposit/src/main/kotlin/com/d3/eth/deposit/endpoint/EthRefundStrategyImpl.kt

```diff
@@ -106,21 +108,24 @@ class EthRefundStrategyImpl(
                     )
                 }
                 // withdrawal case
-                (appearedTx.payload.reducedPayload.commandsCount == 1) &&
-                        commands.hasTransferAsset() -> {
-                    val destAccount = commands.transferAsset.destAccountId
-                    // TODO: Bulat change destAccount to withdrawalTrigger account
-                    if (destAccount != credential.accountId)
-                        throw NotaryException("Refund - check transaction. Destination account is wrong '$destAccount'")
-
-                    val amount = commands.transferAsset.amount
-                    val assetId = commands.transferAsset.assetId
-                    val destEthAddress = commands.transferAsset.description
+                isWithdrawalTransaction(
+                    appearedTx,
+                    withdrawalAccountId
+                ) -> {
+                    // pick withdrawal transfer
+                    val withdrawalCommand = appearedTx.payload.reducedPayload.commandsList
+                        .filter { cmd -> cmd.hasTransferAsset() }
+                        .map { cmd -> cmd.transferAsset }
+                        .first { cmd -> cmd.destAccountId == withdrawalAccountId }
```

Comment:
what if there are many 'withdrawals' within a transaction?

Reply:
Actually now we cannot have many withdrawals within a single transaction since we collect proof for iroha tx hash. So, we assume only one withdrawal for transaction.